### PR TITLE
enable MSVC /MP - for multi core builds of files

### PR DESCRIPTION
**Pull request by RevoluPowered**
TODO: post time comparisons for each build

### CODE REVIEW

File: platform/windows/detect.py [hunk 1]
```diff
@@ -180,37 +180,37 @@ def configure_msvc(env, manual_msvc_config):
...
         env.AppendUnique(CPPDEFINES=["DEBUG_ENABLED"])
-        env.Append(LINKFLAGS=["/DEBUG"])
+        env.Append(LINKFLAGS=["/DEBUG:fastlink"])
 
     env.Append(LINKFLAGS=["/SUBSYSTEM:WINDOWS"])
 
     if env["debug_symbols"] == "full" or env["debug_symbols"] == "yes":
         env.AppendUnique(CCFLAGS=["/Z7"])
-        env.AppendUnique(LINKFLAGS=["/DEBUG"])
+        env.AppendUnique(LINKFLAGS=["/DEBUG:fastlink"])
```
Comment (akien-mga) [inline platform/windows/detect.py:206]: Not sure about this one, `debug_symbols` is used to add debug symbols to release/release_debug binaries, which we still want to optimize as well as we can. What's the impact of `:fastlink` on the resulting binary (size, performance)?

File: platform/windows/detect.py [hunk 2]
```diff
@@ -180,37 +180,37 @@ def configure_msvc(env, manual_msvc_config):
 
     if env["target"] == "release":
         if env["optimize"] == "speed":  # optimize for speed (default)
-            env.Append(CCFLAGS=["/O2"])
+            env.Append(CCFLAGS=["/O2", "/MP"])
```
Comment (akien-mga) [inline platform/windows/detect.py:183]: Factor it out, you apply it in all branches.

File: platform/windows/detect.py [hunk 3]
```diff
@@ -180,37 +180,37 @@ def configure_msvc(env, manual_msvc_config):
...
-        env.AppendUnique(LINKFLAGS=["/DEBUG"])
+        env.AppendUnique(LINKFLAGS=["/DEBUG:fastlink"])
 
     ## Compile/link flags
 
     env.AppendUnique(CCFLAGS=["/MT", "/Gd", "/GR", "/nologo"])
     if int(env["MSVC_VERSION"].split(".")[0]) >= 14:  # vs2015 and later
         env.AppendUnique(CCFLAGS=["/utf-8"])
-    env.AppendUnique(CXXFLAGS=["/TP"])  # assume all sources are C++
+    env.AppendUnique(CXXFLAGS=["/TP", "/MP"])  # assume all sources are C++
```
Comment (akien-mga) [inline platform/windows/detect.py:213]: Isn't it redundant with applying it to `CCFLAGS` (which apply to both C and C++ sources)?

**Comment by RevoluPowered** [comment]
**scons -j16 (branch change to master)**
Size: 144 MB

**scons -j16 (branch change to go_faster_button)**
Size 144 MB

File: platform/windows/detect.py (hunk 2 above)
Comment (RevoluPowered) [reply platform/windows/detect.py:183]: on it

**Comment by RevoluPowered** [comment]
Go Faster Button Branch
real    4m19.372s
user    0m0.000s
sys     0m0.015s


Master 
real    4m3.955s
user    0m0.000s
sys     0m0.015s

**Comment by RevoluPowered** [comment]
useless pr is useless

**Comment by RaTcHeT302** [comment]
did you turn off windows defender? it destroys the compile times, personally i always keep it off from now on, it's more of a hassle than anything

**Pull request closed without merging**